install.sh

```diff
@@ -1,75 +1,57 @@
-#!/bin/sh
+#!/usr/bin/env sh
 # Copyright 2022 khanhas. GPL license.
 # Edited from project Denoland install script (https://github.com/denoland/deno_install)
 
 set -e
 
 case $(uname -sm) in
-"Darwin x86_64") target="darwin-amd64" ;;
-"Darwin arm64") target="darwin-arm64" ;;
-"Linux x86_64") target="linux-amd64" ;;
-*)
-	echo "Unsupported platform $(uname -sm). Only Darwin x86_64, Darwin arm64 and Linux x86_64 binaries are available."
-	exit
-	;;
+	"Darwin x86_64") target="darwin-amd64" ;;
+	"Darwin arm64") target="darwin-arm64" ;;
+	"Linux x86_64") target="linux-amd64" ;;
+	*) echo "Unsupported platform $(uname -sm). Only Darwin x86_64, Darwin arm64 and Linux x86_64 binaries are available."; exit ;;
 esac
 
-if [ $# -eq 0 ]; then
-	latest_release_uri="https://api.github.com/repos/spicetify/spicetify-cli/releases/latest"
-	echo "DOWNLOADING    $latest_release_uri"
-	spicetify_asset_path=$(
-		command curl -sSf "$latest_release_uri" |
-			command grep -o "/spicetify/spicetify-cli/releases/download/.*/spicetify-.*-${target}\\.tar\\.gz" |
-			command head -n 1
-	)
-	if [ ! "$spicetify_asset_path" ]; then exit 1; fi
-	download_uri="https://github.com${spicetify_asset_path}"
-else
-	download_uri="https://github.com/spicetify/spicetify-cli/releases/download/v${1}/spicetify-${1}-${target}.tar.gz"
-fi
+command -v curl >/dev/null || { echo "curl isn't installed\!" >&2; exit 1; }
+command -v tar >/dev/null || { echo "tar isn't installed\!" >&2; exit 1; }
 
-spicetify_install="$HOME/.spicetify"
-path="export PATH=\"\$PATH:\$HOME/.spicetify\""
-
-if (grep -q "bash" $SHELL || [[ -f "$HOME/.bashrc" ]]) && ! grep -q "$path" "$HOME/.bashrc"; then echo "${path}" >>"$HOME/.bashrc" && echo "SAVING         ${spicetify_install} to \$PATH (bash)"; fi
-if (grep -q "zsh" $SHELL || [[ -f "$HOME/.zshrc" ]]) && ! grep -q "$path" "$HOME/.zshrc"; then echo "${path}" >>"$HOME/.zshrc" && echo "SAVING         ${spicetify_install} to \$PATH (zsh)"; fi
-if [[ -f "$HOME/.config/fish/config.fish" ]] && ! grep -q "$path" "$HOME/.config/fish/config.fish"; then echo "${path}" >>"$HOME/.config/fish/config.fish" && echo "SAVING         ${spicetify_install} to \$PATH (fish)"; fi
+# download uri
+shortcut=https://github.com/spicetify/spicetify-cli
+tag=$(curl -LsH 'Accept: application/json' $shortcut/releases/latest)
+tag=${tag%\,\"update_url*}
+tag=${tag##*tag_name\":\"}
+tag=${tag%\"}
+download_uri="$shortcut/releases/download/$tag/spicetify-${tag#v}-$target.tar.gz"
+unset tag
```

Comment:
Aren't we adding extra complexity unnecessarily? We don't need to know the tag

Reply:
is there another way to get the latest release? your script before this had a check for the tag as well.

Reply:
https://github.com/spicetify/spicetify-cli/blob/761ab44ca4d5b4b06b4937f11897b29a6f3c30bc/install.sh#L22

Reply:
But it ignored the tag 🤷 it just went for the latest and that's it! But I guess your approach is actually more readable!